### PR TITLE
PP-407, PP-403: Replace bash with sh in .travis.yml file && Add testdev entry in /etc/hosts during CI

### PR DESCRIPTION
**Pull request by hirenvadalia**
#### Issue-ID

* **[PP-403](https://pbspro.atlassian.net/browse/PP-403)**
* **[PP-407](https://pbspro.atlassian.net/browse/PP-407)**

#### Problem description
* **PP-403:** Travis-CI fails because docker container is not able to resolve short hostname (testdev) with IP as hostname-IP entry for testdev is not added into /etc/hosts. Due to this PBS startup script fails with following error:
```
***
*** Invalid local hostname: testdev
*** This value must resolve to a valid IP address.
***
```
* **PP-407:** Travis-CI fails due to issue in docker engine (see **docker/compose#3379**)

#### Cause / Analysis
* **PP-403:** testdev-IP entry is not added to /etc/hosts file
* **PP-407:** **docker/compose#3379**

#### Solution description
* **PP-403:** Add testdev-IP entry into /etc/hosts file
* **PP-407:** Replaced 'bash' with 'sh' as a workaround suggested by support@travis-ci.com. Also added 'travis_retry' as a prefix to failed command (i.e. cp command) which will try max 3 time if command fails.
* **As this pull request contains fixes for two issue _PP-403_ and _PP-407_ as both fixes requires to pass travis-CI**

#### Checklist:

- [x] I have joined the **[pbspro community forum](http://community.pbspro.org/)**.
- [x] My pull request contains a **single, signed** commit. See **[setting up gpg signature](https://pbspro.atlassian.net/wiki/display/DG/Signing+Your+Git+Commits)**.
- [x] My code follows the **[coding style](https://pbspro.atlassian.net/wiki/display/DG/Coding+Standards)** of this project.
- [ ] My change requires project documentation. See **[required documentation checklist](https://pbspro.atlassian.net/wiki/display/DG/Checklist+for+Developing+Features+and+Bug+Fixes)** for details.
- [ ] I have added documentation in the **[project documentation area](https://pbspro.atlassian.net/wiki/display/PD)**.
- [ ] I have added new **PTL test(s) to my commit**. (See **[using PTL for testing](https://pbspro.atlassian.net/wiki/display/DG/Using+PTL+for+Testing)**) *(or)*
- [ ] I have added  **manual test(s) to the Jira ticket and explained why PTL is not appropriate** for this case.
- [ ] All new and existing automated tests have passed. (See **[running automated PTL tests](https://pbspro.atlassian.net/wiki/display/DG/PTL+Quick+Start+Guide)**).
- [ ] I have attached **test logs to the Jira ticket** as evidence of testing/verification.

__***For further information please visit the [Developer Guide Home](https://pbspro.atlassian.net/wiki/display/DG/Developer+Guide+Home).***__

### CODE REVIEW

File: .travis.yml [hunk 1]
```diff
@@ -9,9 +9,12 @@ before_install:
     - gem install octokit
     - .github/runchecks
     - docker pull ${OS_TYPE}
-    - docker run -it -d -h testdev.pbspro.com --name testdev -v `pwd`:`pwd` --privileged -w `pwd` ${OS_TYPE} /bin/bash
+    - docker run -it -d -h testdev.pbspro.com --name testdev -v `pwd`:`pwd` --privileged -w `pwd` ${OS_TYPE} /bin/sh
     - docker ps -a
     - export DOCKER_EXEC="docker exec -it testdev"
+    - export DOCKER_TESTDEV_IP="$(docker inspect -f '{{ .NetworkSettings.IPAddress }}' testdev)"
+    - ${DOCKER_EXEC} /bin/bash -c "echo '${DOCKER_TESTDEV_IP}     testdev' >> /etc/hosts"
```
Comment (kjakkali) [inline .travis.yml]: @hirenvadalia : Replace bash with sh

Comment (hirenvadalia) [inline .travis.yml]: @kjakkali Thanks for review. I have replaced bash with sh. Please have a look again.

**Comment by kjakkali** [comment]
looks good. I sign off.

**Comment by hirenvadalia** [comment]
@kjakkali Thanks for review and sign-off.

As per process I have squashed all commits into one so that @PBSPro/pbspro-maintainers  can merge these changes.

**Comment by arungrover** [comment]
All looks good to me too.
I sign off

**Comment by arungrover** [comment]
All looks good, merging it in

**Comment by hirenvadalia** [comment]
@arungrover Thanks for merging it.

**Pull request merged**